mmdet/datasets/crowdhuman.py

```diff
@@ -0,0 +1,143 @@
+import json
+import os.path as osp
+from typing import List, Union
+
+import mmcv
+from mmengine.dataset import BaseDataset
+from mmengine.fileio import FileClient
+
+from mmdet.registry import DATASETS
+
+
+@DATASETS.register_module()
+class CrowdHumanDataset(BaseDataset):
+    """Dataset for CrowdHuman."""
+
+    METAINFO = {
+        'CLASSES': ('person', ),
+        # PALETTE is a list of color tuples, which is used for visualization.
+        'PALETTE': [(220, 20, 60)]
+    }
+
+    def __init__(self, data_root, file_client_args: dict = dict(backend='disk'),
+                 **kwargs):
+        assert 'data_root' in kwargs, \
+            f"{self.__class__.__name__} needs parameter: 'data_root' \n"
+        assert osp.exists(kwargs['data_root']), \
+            "'data_root' is not valid: Please enter the correct parameter \n"
+        self.id_hw_path = osp.join(kwargs['data_root'], 'id_hw.json')
+        self.id_hw_exist_flag = True
+        if not osp.isfile(self.id_hw_path):
+            self.file_client = FileClient(**file_client_args)
+            self.id_hw_exist_flag = False
+            self.id_hw = {}
+        else:
+            with open(self.id_hw_path, 'r') as id_hw_file:
+                self.id_hw = json.load(id_hw_file)
+        super().__init__(data_root=data_root, **kwargs)
+
+    def load_data_list(self) -> List[dict]:
+        """Load annotations from an annotation file named as ``self.ann_file``
+
+        Returns:
+            List[dict]: A list of annotation.
+        """  # noqa: E501
+        with open(self.ann_file, 'r') as file:
+            anno_strs = file.readlines()
+
+        print('loading CrowdHuman annotation...')
+        data_list = []
+        prog_bar = mmcv.ProgressBar(len(anno_strs))
+        for i, anno_str in enumerate(anno_strs):
+            anno_dict = json.loads(anno_str)
+            parsed_data_info = self.parse_data_info(anno_dict)
+            data_list.append(parsed_data_info)
+            prog_bar.update()
+        if not self.id_hw_exist_flag:
+            with open(self.id_hw_path, 'w', encoding='utf-8') as file:
+                json.dump(self.id_hw, file, indent=4)
+            print(f'\nsave id_hw in {self.data_root}')
+        del self.id_hw, self.id_hw_exist_flag, self.id_hw_path
+        print('Done')
+        return data_list
+
+    def parse_data_info(self, raw_data_info: dict) -> Union[dict, List[dict]]:
+        """Parse raw annotation to target format.
+
+        Args:
+            raw_data_info (dict): Raw data information load from ``ann_file``
+
+        Returns:
+            Union[dict, List[dict]]: Parsed annotation.
+        """
+        data_info = {}
+        img_path = osp.join(self.data_prefix['img'],
+                            f"{raw_data_info['ID']}.jpg")
+        data_info['img_path'] = img_path
+        data_info['img_id'] = raw_data_info['ID']
+
+        if not self.id_hw_exist_flag:
+            img_bytes = self.file_client.get(img_path)
```

Comment:
```suggestion
            print_log('id_hw file does not exist, start collecting image height and width...', level=logging.INFO)
            img_bytes = self.file_client.get(img_path)
```

Reply:
We use a progress bar display dataset loading progress, in order to more beautiful print output information, the prompt of creating id_wh.json is put in the init.